### PR TITLE
[skip changelog] Update client_example with proxy testing

### PR DESCRIPTION
**Pull request by silvanocerza**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**

Adds some examples to the `client_example` project used to "test" gRPC comunication.

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

File: client_example/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Client example
+
+This a client that simulates a gRPC consumer. We're using this for the time being to test the interaction with the gRPC
+interface, hopefully in the future we'll have proper integration tests.
```
Comment (per1234) [inline client_example/README.md:4]: The purpose of client_example is not testing. The purpose is documentation.

**Review by per1234 — CHANGES_REQUESTED** [review]

**Review by rsora — APPROVED** [review]
As we discussed internally, the original purpose of the client_example core was to provide code and a bit of documentation on how to use the CLI gRPC interface but would like to leverage it  in the future as a component of our integration tests suite, this PR is a step in that direction.

**Comment by per1234** [comment]
I really don't understand why there is this trend of trying to use client_example for a purpose it was never intended for. This already happened once last year and it made it much less useful for the sake of documentation (https://github.com/arduino/arduino-cli/issues/353). I don't see why we can't just write dedicated integration tests and leave the client example as an example.

I'd like to get input on this from the creator of the example: @cmaglie

**Comment by silvanocerza** [comment]
My main concern with the `client_example` is that it's practically never run but to verify that the gRPC has not been broken. Also it's not run automatically so it's often broken without anyone noticing.

Ideally well written tests can, and should, act as documentation for developers. I think right now the `client_example` fails on both since it's not run automatically and it doesn't cover all cases.

My goal is to remove it or adapt it in favour of integration testing, make it run automatically so we know if the gRPC interface has been broken, and also cover more cases.

**Comment by per1234** [comment]
Running client_example automatically is utterly trivial. It would be less work to add it to the workflow than it has been to talk about it.

**Comment by per1234** [comment]
I'm simply not convinced by the test as documentation argument. When writing tests, the sole goal should be to test. When writing documentation the sole goal should be to document. If you try to do both at the same time, you're going to end up needing to make compromises.

**Comment by cmaglie** [comment]
> I'd like to get input on this from the creator of the example: @cmaglie

Heh, it started as a testing program to try out the grpc interface in the very early times but, once the gRPC API has been bootstrapped, to not waste it we transformed this test program into an example of how to use the grpc interface, something like a minimal client stub to tinker or to develop on.

> I'm simply not convinced by the test as documentation argument. When writing tests, the sole goal should be to test. When writing documentation the sole goal should be to document. If you try to do both at the same time, you're going to end up needing to make compromises.

I agree, we have the integration test framework, we should add tests there.

Also, right now we have a single example_client that runs a bunch of random actions, IMHO it would make more sense to split it into smaller examples that show one feature at a time... but I don't know if it will ever happen, I guess this is a very low priority.

**Pull request merged**